### ISSUE TITLE
Make `--target` / `-d` options consistent between vegastrike-engine and vegasettings

### ISSUE THREAD

**Issue by stephengtuggy** [vegastrike/Vega-Strike-Engine-Source#638]
Preferably, either option should work with either program.

**Comment by BenjamenMeyer** [comment]
yes, we need to refactor our source layout and how CMake works...kind of goes hand-in-hand, which would also make this change easier.

**Comment by LifWirser** [comment]
note GTK needs to be made manditory in  Cmake . otherwise it reys to make an old "turbo c/c++" call to 

`void Start(int * argc, char ***argv) {
	LoadMainConfig();
	InitGraphics(argc,argv);
	LoadConfig();
	ShowMain();
`}
the "``InitGraphics(" is the turbo c/c++ bit
Also have you condiered  making these in o "preference" submenu of main vsengine

oh one more thinge a "font" size /... I set up resolution to fit my monitor ans could not read the small text when running
last sorry I'm late had a bout of the scourge/supply issues etc

**Comment by stephengtuggy** [comment]
Use [Boost.Program_options](https://www.boost.org/doc/libs/1_79_0/doc/html/program_options.html), as @BenjamenMeyer mentioned on #581 ?

**Comment by BenjamenMeyer** [comment]
@stephengtuggy yes

**Comment by BenjamenMeyer** [comment]
@stephengtuggy we just ported this forward to 0.9.x and master no?

**Comment by stephengtuggy** [comment]
Finished as of #999
